bokehjs/src/lib/models/formatters/timedelta_tick_formatter.ts

```diff
@@ -0,0 +1,312 @@
+import {ContextWhich, Location, TimedeltaResolutionType} from "core/enums"
+import type * as p from "core/properties"
+import {assert} from "core/util/assert"
+import {sprintf} from "core/util/templating"
+import {isString, isArray, isBoolean, is_undefined} from "core/util/types"
+import type {Arrayable} from "core/types"
+import {TickFormatter} from "models/formatters/tick_formatter"
+import {ONE_DAY, ONE_HOUR, ONE_MICRO, ONE_MILLI, ONE_MINUTE, ONE_NANO, ONE_SECOND} from "models/tickers/util"
+
+export type {TimedeltaResolutionType} from "core/enums"
+
+// Labels of time units, from finest to coarsest.
+export const resolution_order: TimedeltaResolutionType[] = [
+  "nanoseconds", "microseconds", "milliseconds", "seconds", "minsec", "minutes", "hourmin", "hours", "days",
+]
+
+export const formatting_map: {[template: string]: any} = {
+  "%Nano": (t: number): string => _ns(t, 1_000),
+  "%nano": (t: number): string => _ns(t, null),
+  "%Micro": (t: number): string => _us(t, 1_000),
+  "%micro": (t: number): string => _us(t, null),
+  "%Milli": (t: number): string => _ms(t, 1_000),
+  "%milli": (t: number): string => _ms(t, null),
+  "%Second": (t: number): string => _seconds(t, 60),
+  "%second": (t: number): string => _seconds(t, null),
+  "%Minute": (t: number): string => _minutes(t, 60),
+  "%minute": (t: number): string => _minutes(t, null),
+  "%Hour": (t: number): string => _hours(t, 24),
+  "%hour": (t: number): string => _hours(t, null),
+  "%day": (t: number): string => _days(t, null),
```

Comment:
I haven't seen anything like these format codes anywhere else. Were they derived from some reference you can point to? Offhand, I would say that these codes should match the datetime tick formatter codes as much as possible where they overlap or have very similar use, and then for any that don't, try to be a similar style.